site/content/docs/howto/configure-supervisor-with-auth0.md

```diff
@@ -0,0 +1,147 @@
+The examples below are provided as examples to better understand how Auth0 and Pinniped integrate, not how to configure Auth0.
+
+Assuming that you have somehow configured Auth0 to include group membership information about your users, you can expose this to Pinniped by configuring Auth0 to include a [custom claim](https://auth0.com/blog/adding-custom-claims-to-id-token-with-auth0-actions/) in the Auth0 ID token.
+
+Auth0 recommends using a [namespaced format](https://auth0.com/docs/secure/tokens/json-web-tokens/create-custom-claims) for your custom claim names.
+In the following example, replace `"https://example.com/pinniped/groups"` with the namespaced claim name of your choice.
+Pinniped requires that the value of the group claim is an array of strings.
+
+The following example is intended to show how to add a custom claim, but does not show a realistic example of where the group names should come from.
+To keep this example simple, the group names shown here are hardcoded.
+Do not hardcode group names for a production system.
+Instead, the array of groups should be dynamically provisioned from the appropriate place in the Auth0 user store.
+
+```typescript
+exports.onExecutePostLogin = async (event, api) => {
+  if (event.authorization) {
+    api.idToken.setCustomClaim("https://example.com/pinniped/groups", ["auth0-read-only", "other-grouo", "something-else"]);
+  }
+};
+```
+
+To configure your Kubernetes authorization, please see [how-to login]({{< ref "login" >}}).
+
+## Configure the Supervisor
+
+Create an [OIDCIdentityProvider](https://github.com/vmware-tanzu/pinniped/blob/main/generated/{{< latestcodegenversion >}}/README.adoc#oidcidentityprovider) in the same namespace as the Supervisor.
+
+For example, this OIDCIdentityProvider uses Auth0's `email` claim as the Kubernetes username:
+
+```yaml
+apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+kind: OIDCIdentityProvider
+metadata:
+  namespace: pinniped-supervisor
+  name: auth0
+spec:
+
+  # Change this to be the actual issuer provided by your Auth0 account.
+  issuer: https://<your-tenant-id>.<your-region>.auth0.com/
+
+  authorizationConfig:
+
+    # Request any scopes other than "openid" for claims besides
+    # the default claims in your token. The "openid" scope is always
+    # included.
+    additionalScopes: 
+      - offline_access
+      - email
+
+    # If you would also like to allow your end users to authenticate using
+    # a password grant, then change this to true. Password grants only work
+    # with applications created in Auth0 with the "Password" grant type enabled.
+    allowPasswordGrant: false
+
+  # Specify how Auth claims are mapped to Kubernetes identities.
+  claims:
+
+    # Specify the name of the claim in your Auth0 ID token that will be mapped
+    # to the "username" claim in downstream tokens minted by the Supervisor.
+    username: email
+
+    # Specify the name of the claim in your Auth0 ID token that represents the groups
```

Comment:
This wraps a little funny in the final rendered site.  Maybe we can split the line?
<img width="868" alt="Screenshot 2023-07-28 at 3 31 37 PM" src="https://github.com/vmware-tanzu/pinniped/assets/280512/cea2ef6b-19c9-49a6-b7fb-037e017c1245">

Reply:
done, see next push